src/lib/abstractimagedownloader.cpp

```diff
@@ -305,9 +313,34 @@ QNetworkReply *AbstractImageDownloader::createReply(const QString &url, const QV
     return d->networkAccessManager->get(request);
 }
 
+static QString createOutputPath(SocialSyncInterface::DataType dataType,
+                                SocialSyncInterface::SocialNetwork socialNetwork,
+                                const QString &subdir,
+                                const QString &identifier,
+                                const QString &mimetype)
+{
+    QString result = QString(PRIVILEGED_DATA_DIR) + QChar('/') + SocialSyncInterface::dataType(dataType) + QChar('/');
+
+    if (dataType == SocialSyncInterface::Contacts) {
+        result += QStringLiteral("avatars") + QChar('/');
+    }
+
+    result += SocialSyncInterface::socialNetwork(socialNetwork) + QChar('/')
+            + subdir + QChar('/');
+
+    // do we want to support more image types?
+    if (mimetype == QStringLiteral("image/png")) {
+        result += identifier + QStringLiteral(".png");
+    } else {
+        result += identifier + QStringLiteral(".jpg");
```

Comment:
"result += identifier" == common to all branches -> could be done outside the conditionals

Also wondering whether it would be worthwhile to handle image/jpeg  and fallback jpeg separately - like perhaps emit warning in the latter case? [and related to that, in calling code elsewhere using empty mimetype string vs stating that save as jpeg is expected]

Reply:
Would work that way too, but filename is kind of a unit so thought it would be more understandable having it constructed as such.

The fallback shouldn't do warnings for now. It's actually the base case from earlier behavior.